src/components/internals/calendar-body/calendar-body.styles.js

```diff
@@ -18,32 +18,30 @@ const getClearSectionStyles = props => {
     padding: ${vars.spacingXs};
     cursor: pointer;
     outline: 0;
+    transition: color ${vars.transitionStandard},
+      border-color ${vars.transitionStandard};
   `;
-  if (props.isDisabled) {
+  if (props.isOpen || props.isFocused) {
```

Comment:
We're not rendering ClearSection when `isDisabled`.
And this other style that I added was missing.